src/components/transactionsV2/transactionRowV2.js

```diff
@@ -50,32 +52,46 @@ class TransactionRowV2 extends React.Component {
 
   render() {
     const { props } = this;
+    const { value } = props;
     const onClick = props.onClick || (() => {});
     const hasConfirmations = props.value.confirmations && props.value.confirmations > 0;
     const { isConfirmed } = this.state;
     return (
       <TableRow className={`${grid.row} ${styles.row} ${!hasConfirmations ? styles.pending : ''} transactions-row`} onClick={() => onClick(props)}>
         <div className={`${grid['col-sm-4']} ${grid['col-lg-3']} transactions-cell`}>
-          <TransactionTypeV2 {...props.value}
+          <Icon name={props.address === value.senderId ? 'outgoing' : 'incoming' } className={styles.inOutIcon} />
+          <TransactionTypeFigure
+            address={props.address === value.senderId ? value.recipientId : value.senderId }
+            transactionType={value.type}
+          />
+          <TransactionAddress
+            address={value.recipientId}
             bookmarks={props.bookmarks}
-            address={props.address} />
+            t={props.t}
+            token={value.token}
+            transactionType={value.type}
+          />
         </div>
-          <div className={`${grid['col-sm-3']} ${grid['col-lg-3']} transactions-cell`}>
-            <TransactionDetailV2
-              t={props.t}
-              transaction={props.value} />
-          </div>
         <div className={`${grid['col-sm-2']} ${grid['col-lg-2']} transactions-cell`}>
           <div className={`${styles.status} ${!isConfirmed ? styles.showSpinner : styles.showDate}`}>
             <SpinnerV2 completed={hasConfirmations} label={props.t('Pending...')} />
             <DateTimeFromTimestamp time={props.value.timestamp} token={props.value.token} />
           </div>
         </div>
         <div className={`${grid['col-sm-1']} ${grid['col-lg-2']} transactions-cell`}>
-          <LiskAmount val={props.value.fee}/>&nbsp;{props.value.token || props.t('LSK')}
+          <LiskAmount val={props.value.fee}/>&nbsp;{props.value.token}
```

Comment:
The token shouldn't be part of transaction list data. You must get from setting store if it is needed

Reply:
I think we both want to avoid unnecessary extra complexity and having a token part of the transaction object causes less complexity than having an extra token prop on all components that render transaction amount or have such child. We already have a function the modifies each transaction object for BTC https://github.com/LiskHQ/lisk-hub/blob/b51ca76a4f5306fbff57074ef727571024f79eca/src/utils/api/btc/transactions.js#L15 , so IMO it is quite clean to have a similar function also for LSK (and for potential future tokens).

### If the token is part of the transaction object, these files need to have extra complexity:
- https://github.com/LiskHQ/lisk-hub/blob/b51ca76a4f5306fbff57074ef727571024f79eca/src/utils/api/lsk/transactions.js
- https://github.com/LiskHQ/lisk-hub/blob/b51ca76a4f5306fbff57074ef727571024f79eca/src/utils/api/btc/transactions.js

### If the token is passed separately into all components, these files need to have extra complexity (extra prop):
- https://github.com/LiskHQ/lisk-hub/blob/b51ca76a4f5306fbff57074ef727571024f79eca/src/components/singleTransactionV2/index.js
- https://github.com/LiskHQ/lisk-hub/blob/b51ca76a4f5306fbff57074ef727571024f79eca/src/components/singleTransactionV2/singleTransactionV2.js
- https://github.com/LiskHQ/lisk-hub/blob/b51ca76a4f5306fbff57074ef727571024f79eca/src/components/transactionsV2/transactionDetailViewV2/transactionDetailViewV2.js
- https://github.com/LiskHQ/lisk-hub/blob/b51ca76a4f5306fbff57074ef727571024f79eca/src/components/transactionsV2/amountV2.js
- https://github.com/LiskHQ/lisk-hub/blob/b51ca76a4f5306fbff57074ef727571024f79eca/src/components/transactionsV2/transactionRowV2.js

To sum up, they are two different approaches to prevent some complexity in the code, each has some pros and cons. The differences are not super huge, so I suggest to keep the already implemented approach as it involves no extra work.

Reply:
First of all, by adding the active token to each transaction row you store data which is dynamic in two different places. Then, every time the active token changes you need to up two different places.
On the other hand, the whole concept of global state management was introduced to fix the problem that you encounter. So you there is no need to inject active token into each transaction row because as soon as a user changes the active token the whole transaction list will be fetched again.

Reply:
> First of all, by adding the active token to each transaction row you store data which is dynamic in two different places. Then, every time the active token changes you need to up two different places.

This is not true at all. Token of a transaction is not dynamic. If active token changes, an LSK transaction is still LSK transaction. I don't need to update anything extra.

> On the other hand, the whole concept of global state management was introduced to fix the problem that you encounter. So you there is no need to inject active token into each transaction row because as soon as a user changes the active token the whole transaction list will be fetched again.

This is also not true at all. The problem is not about state management, state management works the same no matter if I use activeToken or token inside transaction. If I don't pass token (LSK/BTC) into each row, I cannot display that the amount of the transaction is in LSK or BTC.
<img width="1062" alt="Screenshot 2019-06-05 at 10 56 37" src="https://user-images.githubusercontent.com/1254342/58943828-df188300-8780-11e9-99ae-d941d724bd73.png">

Reply:
@yasharAyari, on a more personal note, reading your comment above makes me feel like you're telling me what to do while you have no clue what you're talking about, which makes me quite upset. I don't mean to offend you, but I want to express how I feel about this discussion.